storage/cached_rawstore.go

```diff
@@ -94,9 +96,9 @@ func (s *cachedRawStore) WriteRaw(ctx context.Context, reference DataReference,
 	err = s.cache.Set([]byte(reference), buf.Bytes(), neverExpire)
 	if err != nil {
 		s.metrics.CacheWriteError.Inc()
+		err = errors.Wrapf(ErrFailedToWriteCache, err, "Failed to Cache the metadata")
```

Comment:
so we are returning an error, but we ignore it at higher levels. Isnt the effect the same as the current code? How was the current code failing?

Reply:
It was not being ignored by higher levels... we were actually failing if we succeed to write to S3 but fail to write to this cache... the PR adds a specific error code so that higher levels can know what to check for.

Reply:
This function, WriteRaw(), could fail because of other reasons. For example, it could fail because of the failure of `s.RawStore.WriteRaw()`, which should not be and is not ignored at the higher level in this implementation.